### PR TITLE
fix :  refactor quota  table

### PR DESCRIPTION
**Pull request by flywukong**
### Description

1. fix the return value of getUpdatedConsumedQuota function,  the order is wrong now
2. refactor quota table to support better get quota computing. The original way failed to get the right value of the consumed free quota of this month , and  need to query the table of last month to compute the consumed free quota of this month which cause more db operation
image

### Rationale

fix quota logic

### Example

```
[root@ localtest]# ./gnfd-cmd --passwordfile password.txt bucket get-quota gnfd://testbucket4
quota info:
 charged quota:4476395008 
 free quota:1344556800    // remained free quota
 consumed quota:0 
 free consumed quota: 4429185024 
[root@ localtest]# ./gnfd-cmd --passwordfile password.txt object get gnfd://testbucket4/dss5ds download2
download object dss5ds, the file path is download2, content length:1476395008 
[root localtest]# ./gnfd-cmd --passwordfile password.txt bucket get-quota gnfd://testbucket4
quota info:
 charged quota:4476395008 
 free quota:0        // remained free quota
 consumed quota:131838208   // consumed charged quota
 free consumed quota: 5773741824 

[root@tf_nodereal_dev_storage_QA_testing2_ec2 localtest]# ./gnfd-cmd --passwordfile password.txt object get gnfd://testbucket4/dss5ds download2
download object dss5ds, the file path is download2, content length:1476395008 
[root@tf_nodereal_dev_storage_QA_testing2_ec2 localtest]# ./gnfd-cmd --passwordfile password.txt bucket get-quota gnfd://testbucket4
quota info:
 charged quota:4476395008 
 free quota:0 
 consumed quota:1608233216  // consumed charged quota
 free consumed quota: 5773741824 
```
### Changes

Notable changes: 
* add each change in a bullet point here
* ...

### CODE REVIEW

File: store/sqldb/traffic.go [hunk 1]
```diff
@@ -79,30 +79,40 @@ func (s *SpDBImpl) CheckQuotaAndAddReadRecord(record *corespdb.ReadRecord, quota
...
 
-func getUpdatedConsumedQuota(record *corespdb.ReadRecord, freeQuota, freeConsumedQuota, totalConsumeQuota, chargedQuota uint64) (uint64, uint64, error) {
-	recordQuotaCost := record.ReadSize
-	needCheckChainQuota := true
-	freeQuotaRemain := freeQuota - freeConsumedQuota
-	// if remain free quota more than 0, consume free quota first
+// getUpdatedConsumedQuota compute the updated quota of traffic table by the incoming read cost and the newest record.
+// it returns the updated consumed free quota,consumed charged quota and remained free quota
+func getUpdatedConsumedQuota(recordQuotaCost, freeQuotaRemain, consumeFreeQuota, consumeChargedQuota, chargedQuota uint64) (uint64, uint64, uint64, bool, error) {
+	// if remain free quota more than 0 and enough, just consume free quota
```
Comment (RenRick) [inline store/sqldb/traffic.go:85]:
```suggestion
	// if remain free quota enough, just consume free quota
```

**Review by mliu233 — APPROVED** [review]

File: modular/metadata/metadata_bucket_service.go [hunk 2]
```diff
@@ -347,6 +347,7 @@ func (r *MetadataModular) GfSpGetBucketReadQuota(
 				", bucket_id: " + req.GetBucketInfo().Id.String() + ", error: " + err.Error())}, nil
 		}
 	}
+
```
[the remark this answers is not in the document]
Comment (flywukong) [reply modular/metadata/metadata_bucket_service.go]: fixed

File: store/sqldb/traffic.go (hunk 1 above)
Comment (flywukong) [reply store/sqldb/traffic.go]: fixed

File: store/sqldb/traffic.go [hunk 3]
```diff
@@ -79,30 +79,40 @@ func (s *SpDBImpl) CheckQuotaAndAddReadRecord(record *corespdb.ReadRecord, quota
...
+	needUpdateFreeQuota := false
 	if freeQuotaRemain > 0 && recordQuotaCost < freeQuotaRemain {
```
[the remark this answers is not in the document]
Comment (flywukong) [reply store/sqldb/traffic.go]: fixed

**Review by annielz — APPROVED** [review]

**Review by BarryTong65 — APPROVED** [review]

**Review by RenRick — APPROVED** [review]
LGTM

**Pull request merged**